src/shared/spawn-polkit-agent.c

```diff
@@ -42,11 +42,19 @@ int polkit_agent_open(void) {
 
         xsprintf(notify_fd, "%i", pipe_fd[1]);
 
+        if (access(POLKIT_AGENT_BINARY_PATH, X_OK) < 0)
+                return log_error_errno(errno, "Failed to find polkit agent binary. Is polkit agent installed?");
+
         r = fork_agent("(polkit-agent)",
-                       &pipe_fd[1], 1,
+                       &pipe_fd[1],
+                       1,
                        &agent_pid,
                        POLKIT_AGENT_BINARY_PATH,
-                       POLKIT_AGENT_BINARY_PATH, "--notify-fd", notify_fd, "--fallback", NULL);
+                       POLKIT_AGENT_BINARY_PATH,
+                       "--notify-fd",
+                       notify_fd,
+                       "--fallback",
+                       NULL);
```

Comment:
Hmm, not your fault of course, but below we log something spurious: `Failed to fork TTY ask password agent: %m`

If you want to improve the logging could you please fix that too?

Reply:
ack